### PR TITLE
ddl: args v2 for Alter/Remove Partitioning and Add/Drop/Reorganize Partition

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #53930

Problem Summary:

### What changed and how does it work?
as title
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/ddl/rollingback.go [hunk 1]
```diff
@@ -340,18 +340,18 @@ func convertAddTablePartitionJob2RollbackJob(jobCtx *jobContext, t *meta.Meta, j
...
-		if err != nil {
-			return ver, err
-		}
-		job.Args = []any{partNames, partInfo}
+		args.PartNames = partNames
+		job.FillArgs(args)
 	} else {
-		job.Args = []any{partNames}
+		args.PartNames = partNames
+		model.FillRollbackArgsForAddPartition(job, args)
```
Comment (joccau) [inline pkg/ddl/rollingback.go:354]: Why not job.FillArgs(args) directly ?

**Review by joccau — COMMENTED** [review]
rest lgtm.

File: pkg/ddl/rollingback.go (hunk 1 above)
Comment (D3Hunter) [reply pkg/ddl/rollingback.go:354]: see comments on TablePartitionArgs, it changes param order and type in this case

**Review by joccau — APPROVED** [review]
LGTM

**Review by lance6716 — APPROVED** [review]

**Pull request merged**